vcr/request.py

```diff
@@ -32,9 +33,19 @@ def __init__(self, method, uri, body, headers):
             self.body = body.read()
         else:
             self.body = body
-        self.headers = {}
-        for key in headers:
-            self.add_header(key, headers[key])
+        self.headers = CaseInsensitiveDict()
```

Comment:
> seems orthogonal to this PR. :-)

Not sure I completely agree. This is a significant (at least minor version bump) change to the behavior of headers. I'd rather not have to do ANOTHER minor version bump for more breaking changes to the way that headers behave.

I suppose that it could be in a separate pr before the minor version bump, but I think that now seems like a good time to handle this to me.